File: SecureDigitalHealthcare/Utilities/AppEncryptor.cs
using System.Security.Cryptography;
using System.Text;

namespace SecureDigitalHealthcare.Utilities
{

    public class AppEncryptor
    {
        private const string encryptionKey = "7RBOwUpx1cv7VR+Bi3tWyI+QkWwC5NN7";

        public static byte[] EncryptBytes(byte[] plainBytes)
        {
            using (Aes aesAlg = Aes.Create())
            {
                aesAlg.Key = Encoding.UTF8.GetBytes(encryptionKey);
                aesAlg.IV = new byte[16]; // Initialization vector

                // Create an encryptor To perform the stream transform
                ICryptoTransform encryptor = aesAlg.CreateEncryptor(aesAlg.Key, aesAlg.IV);

                // Create the streams used for encryption
                using (MemoryStream msEncrypt = new MemoryStream())
                {
                    using (CryptoStream csEncrypt = new CryptoStream(msEncrypt, encryptor, CryptoStreamMode.Write))
                    {
                        csEncrypt.Write(plainBytes, 0, plainBytes.Length);
                        csEncrypt.FlushFinalBlock();
                        return msEncrypt.ToArray();
                    }
                }
            }
        }
        public static byte[] DecryptBytes(byte[] encryptedBytes)
        {
            using (Aes aesAlg = Aes.Create())
            {
                aesAlg.Key = Encoding.UTF8.GetBytes(encryptionKey);
                aesAlg.IV = new byte[16]; // Initialization vector

                // Create a decryptor To perform the stream transform
                ICryptoTransform decryptor = aesAlg.CreateDecryptor(aesAlg.Key, aesAlg.IV);

                // Create the streams used for decryption
                using (MemoryStream msDecrypt = new MemoryStream(encryptedBytes))
                {
                    using (CryptoStream csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read))
                    {
                        using (MemoryStream decryptedStream = new MemoryStream())
                        {
                            csDecrypt.CopyTo(decryptedStream);
                            return decryptedStream.ToArray();
                        }
                    }
                }
            }
        }

        public static string EncryptString(string plainText)
        {
            byte[] plainBytes = Encoding.UTF8.GetBytes(plainText);
            byte[] encryptedBytes = EncryptBytes(plainBytes);
            return Convert.ToBase64String(encryptedBytes);
        }

        public static string DecryptString(string encryptedText)
        {
            byte[] encryptedBytes = Convert.FromBase64String(encryptedText);
            byte[] decryptedBytes = DecryptBytes(encryptedBytes);
            return Encoding.UTF8.GetString(decryptedBytes);
        }

        public static string DecryptString(string encryptedText, string ivString)
        {
            try
            {
                byte[] encryptedBytes = Convert.FromBase64String(encryptedText);
                byte[] keyBytes = Encoding.UTF8.GetBytes(encryptionKey);
                byte[] ivBytes = Convert.FromBase64String(ivString); // Decode Base64 IV

                if (ivBytes.Length != 16)
                {
                    throw new ArgumentException("Invalid IV length");
                }

                using (Aes aesAlg = Aes.Create())
                {
                    aesAlg.Key = keyBytes;
                    aesAlg.IV = ivBytes;
                    aesAlg.Padding = PaddingMode.PKCS7; // Ensure PKCS7 padding

                    // Create a decryptor to perform the stream transform
                    ICryptoTransform decryptor = aesAlg.CreateDecryptor(aesAlg.Key, aesAlg.IV);

                    using (MemoryStream msDecrypt = new MemoryStream(encryptedBytes))
                    {
                        using (CryptoStream csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read))
                        {
                            using (StreamReader srDecrypt = new StreamReader(csDecrypt))
                            {
                                return srDecrypt.ReadToEnd();
                            }
                        }
                    }
                }
            }
            catch (FormatException e)
            {
                Console.WriteLine("Base64 decoding failed: " + e.Message);
                throw;
            }
            catch (CryptographicException e)
            {
                Console.WriteLine("Decryption failed: " + e.Message);
                throw;
            }
        }

    }
}

File: EasyHealthVideoCall/appEncryptor.js

const CryptoJS = require('crypto-js');

class MyAppEncryptor {
    constructor() {
        this.encryptionKey = CryptoJS.enc.Utf8.parse("7RBOwUpx1cv7VR+Bi3tWyI+QkWwC5NN7");
        this.iv = CryptoJS.lib.WordArray.random(16); // Initialization vector
    }

    getIv() {
        return this.iv.toString();
    }
    getIvBase64() {
        return CryptoJS.enc.Base64.stringify(this.iv)
    }

    encryptString(plainText) {
        const encrypted = CryptoJS.AES.encrypt(plainText, this.encryptionKey, {
            iv: this.iv,
            mode: CryptoJS.mode.CBC,
            padding: CryptoJS.pad.Pkcs7 // Ensure PKCS7 padding
        });

        console.log("Plain text: " + plainText)
        console.log("Encrypted: " + encrypted)
        console.log("Encryption key: " + this.encryptionKey);
        console.log("IV: " + this.getIv() + " length: " + this.getIv().length);
        console.log("IV Base64: " + this.getIvBase64() + " length: " + this.getIvBase64().length);

        return encrypted.toString();
    }

    decryptString(encryptedText) {
        const decrypted = CryptoJS.AES.decrypt(encryptedText, this.encryptionKey, {
            iv: this.iv,
            mode: CryptoJS.mode.CBC,
            padding: CryptoJS.pad.Pkcs7
        });

        return decrypted.toString(CryptoJS.enc.Utf8);
    }
}

function encryptString(plainText) {
    const encrypted = CryptoJS.AES.encrypt(plainText, encryptionKey, { iv });
    return encrypted.toString();
}

function decryptString(encryptedText) {
    const decrypted = CryptoJS.AES.decrypt(encryptedText, encryptionKey, { iv });
    return decrypted.toString(CryptoJS.enc.Utf8);
}

// Exporting functions
module.exports = {
    MyAppEncryptor
};